doc/security/vulnerabilities.rst

```diff
@@ -11,15 +11,380 @@ sufficient time to fix them.  Because the release notes are locked to
+
+- `Zephyr project bug tracker ZEPSEC-18
+  <https://zephyrprojectsec.atlassian.net/browse/ZEPSEC-18>`_
+
+- `PR13048 fix for v1.14.0
+  <https://github.com/zephyrproject-rtos/zephyr/pull/13048>`_
+
+CVE-2019
+========
+
+CVE-2019-9506
+-------------
+
+The Bluetooth BR/EDR specification up to and including version 5.1
+permits sufficiently low encryption key length and does not prevent an
+attacker from influencing the key length negotiation. This allows
+practical brute-force attacks (aka "KNOB") that can decrypt traffic
+and inject arbitrary ciphertext without the victim noticing.
+
+- `CVE-2019-9506 <http://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2019-9506>`_
+
+- `Zephyr project bug tracker ZEPSEC-20
+  <https://zephyrprojectsec.atlassian.net/browse/ZEPSEC-20>`_
+
+- `PR18702 fix for v1.14.0
+  <https://github.com/zephyrproject-rtos/zephyr/pull/18702>`_
+
+- `PR18659 fix for v2.0.0
+  <https://github.com/zephyrproject-rtos/zephyr/pull/18659>`_
+
+CVE-2020
+========
+
+CVE-2020-10019
+--------------
+
+Buffer Overflow vulnerability in USB DFU of zephyr allows a USB
+connected host to cause possible remote code execution.
+
+This has been fixed in releases v1.14.2, v2.2.0, and v2.1.1.
+
+- `CVE-2020-10019 <http://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2020-10019>`_
+
+- `Zephyr project bug tracker ZEPSEC-25
+  <https://zephyrprojectsec.atlassian.net/browse/ZEPSEC-25>`_
+
+- `PR23460 fix for 1.14.x
+  <https://github.com/zephyrproject-rtos/zephyr/pull/23460>`_
+
+- `PR23457 fix for 2.1.x
+  <https://github.com/zephyrproject-rtos/zephyr/pull/23457>`_
+
+- `PR23190 fix in 2.2.0
+  <https://github.com/zephyrproject-rtos/zephyr/pull/23190>`_
+
+CVE-2020-10021
+--------------
+
+Out-of-bounds write in USB Mass Storage with unaligned sizes
+
+Out-of-bounds Write in the USB Mass Storage memoryWrite handler with
+unaligned Sizes.
+
+See NCC-ZEP-024, NCC-ZEP-025, NCC-ZEP-026
+
+This has been fixed in releases v1.14.2, and v2.2.0.
+
+- `CVE-2020-10021 <https://docs.zephyrproject.org/latest/security/vulnerabilities.html#CVE-2020-10021>`_
+
+- `Zephyr project bug tracker ZEPSEC-26
+  <https://zephyrprojectsec.atlassian.net/browse/ZEPSEC-26>`_
+
+- `PR23455 fix for v1.14.2
+  <https://github.com/zephyrproject-rtos/zephyr/pull/23455>`_
+
+- `PR23456 fix for the v2.1 branch
+  <https://github.com/zephyrproject-rtos/zephyr/pull/23456>`_
+
+- `PR23240 fix for v2.2.0
+  <https://github.com/zephyrproject-rtos/zephyr/pull/23240>`_
+
+CVE-2020-10022
+--------------
+
+UpdateHub Module Copies a Variable-Size Hash String Into a Fixed-Size Array
+
+A malformed JSON payload that is received from an UpdateHub server may
+trigger memory corruption in the Zephyr OS. This could result in a
+denial of service in the best case, or code execution in the worst
+case.
+
+See NCC-ZEP-016
+
+This has been fixed in the below pull requests for master, branch from
+v2.1.0, and branch from v2.2.0.
```

Comment:
This information is not consistent throughout the document, there are other issues with pr for 2.1 that are not mentioned.

Reply:
Are they part of this security issue itself? These notes are really about giving enough information for someone to know if they need to apply the fix.

Reply:
Ohh, I see now your point. These pr were merged, so it is fine :)